lenovo/legion/16ach6h/default.nix

```diff
@@ -7,12 +7,32 @@
     ../../../common/gpu/nvidia/prime.nix
     ../../../common/pc/laptop
     ../../../common/pc/laptop/ssd
+    ./edid
   ];
 
-  hardware.nvidia.prime = {
-    amdgpuBusId = "PCI:6:0:0";
-    nvidiaBusId = "PCI:1:0:0";
-  };
+  config = lib.mkMerge [
+    {
+      hardware.amdgpu.loadInInitrd = lib.mkDefault false;
 
-  services.thermald.enable = lib.mkDefault true;
-}
+      hardware.nvidia = {
+        modesetting.enable = lib.mkDefault true;
+        powerManagement.enable = lib.mkDefault true;
+      };
+      
+      services.thermald.enable = lib.mkDefault true;
+
+      specialisation.ddg.configuration = {
```

Comment:
The specialization thing is nice but is this something you want to have enabled by default? Afaik it will slow down nixos evaluation by factor 2 (and increase memory usage). I think this could be maybe put into its own module and mentioned in the README for people that rely on this?

Reply:
I was not aware evaluation time took such a hit with specialisations. Going by that it is probably better to leave it as separate modules and let the end user set them up if they don't mind the extra evaluation time.

I'd say having the hybrid mode (iGPU and dGPU available) as the default and dedicated mode/DDG (only dGPU) as an "addition" on top would be the way to go.

Reply:
Perhaps it is indeed possible to separate the versions with specialization, but in actual use, we will toggle DDG very frequently.  So I think it is necessary to use the version with specialization as the default. Maybe we can provide a version without specialization and mark it in README?

Reply:
I just added standalone hybrid only and nvidia only (DDG only) modules, and reminded that specialization may be slow down NixOS evaluation and increase memory usage at README of 16ach6h.